[YANG QINGRU] iP

caroline1233456 opened the pull request:
# Parakeet
> “Your mind is for having ideas, not holding them.” – David Allen

Parakeet is your assistant. It can help you remember and manage the things you need to do. It's

* text-based
* ~~user-friendly~~ **SUPER** user-friendly
*  **smart** and **cute**

All you need to do is,

1. Download it from [parakeet](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html)
2. Run it
3. Tell it your tasks
4. Let it manage your tasks for you

And it is <ins>  **FREE!!!!!** </ins>

Features:
- [x] Manage your Todo list
- [x] Manage your deadline
- [x] Manage your event
- [ ] Replace ChatGPT(I believe it will someday in the future :speak_no_evil:)

If you are a NUS student enrolled in CS2103T, you can use it to practice Java. Here is the `main` method:

```
  public static void main(String[] args) {
        new Parakeet().run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

carolineYQR committed:
level0 completed

carolineYQR committed:
level 1 completed

carolineYQR committed:
level 1 modified

carolineYQR committed:
level 2 completed

carolineYQR committed:
level 2 modified

carolineYQR committed:
level 3 completed

carolineYQR committed:
level 4 completed

carolineYQR committed:
Level-4 debug

carolineYQR committed:
test and debug with A-TextUiTesting

carolineYQR committed:
do A-TextUiTesting2 and debug

carolineYQR committed:
Level 5 completed

carolineYQR committed:
level 6 completed

carolineYQR committed:
modify according to OOP

carolineYQR committed:
modify the code, create a class called TaskList

carolineYQR committed:
complete feature for leve 7

carolineYQR committed:
Complete level 7

carolineYQR committed:
Merge branch 'branch-Level-7'
merge branch-Level-7 to master as Level-7 is completed.

carolineYQR committed:
complete level 8

carolineYQR committed:
complete level-8 and modify level -8 feature

carolineYQR committed:
Merge branch 'branch-Level-8'
merge branch-Level-8 to master branch.

carolineYQR committed:
complete feature for A-MoreOOP.

carolineYQR committed:
test OOP structure

carolineYQR committed:
Merge branch 'MoreOOP'
complete new change:More OOP.

carolineYQR committed:
complete A-Packages

carolineYQR committed:
Merge branch 'add-gradle-support'

carolineYQR committed:
complete A-gradle

carolineYQR committed:
change the structure of class Parakeet and Storage

carolineYQR committed:
complete A-JUnit

carolineYQR committed:
Create JavaDoc for most of the public method.

carolineYQR committed:
Merge branch 'branch-A-JavaDoc'
merge branch-JavaDoc to master

carolineYQR committed:
complete Level-9

carolineYQR committed:
Adjust naming and identation to comply with the coding standard.

carolineYQR committed:
Merge branch 'branch-A-CodingStandard'
merge branch-A-CodingStandard to master

carolineYQR committed:
Merge Level-9 feature to master branch

carolineYQR committed:
Test find feature

carolineYQR committed:
Add JavaDoc for public method inclass Ui

carolineYQR committed:
Fix code style issues

carolineYQR committed:
Add a GUI to the chatbot

carolineYQR committed:
Add GUI to the chatbot

carolineYQR committed:
Merge branch 'branch-Level-10'
Merge branch-Level-10 to master

mjyan02 commented on src/main/java/parakeet/Parakeet.java:
```diff
@@ -0,0 +1,74 @@
+package parakeet;
```
I like how you named all your packages in all lower case, following the basic java coding standards.

mjyan02 commented on src/main/java/parakeet/Parakeet.java:
```diff
@@ -0,0 +1,74 @@
+ * @version 1.0
+ * @since 2025-01-29
+ */
+public class Parakeet {
```
I like how all your Class/enum names are nouns and written in PascalCase while variable names are in camelCase, following basic java coding standards.

mjyan02 commented on src/main/java/parakeet/Parakeet.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+
+    public String getResponse(String input) {
```
I like how your names representing methods are verbs and written in camelCase, following basic java coding standards.

mjyan02 commented on src/main/java/parakeet/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+     *
+     * @param index the index in taskList.
+     */
+    public void complete(int index) {
```
Will it be better to have this method name as mark and the below method name (for unDone) as unMark? This might be a more intuitive name since both methods do the opposite of each other.

mjyan02 commented on src/main/java/parakeet/Storage.java:
```diff
@@ -0,0 +1,131 @@
+        while(this.scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+
+            String[] components = line.split("\\|");
```
I like how you used plural form on names representing a collection of objects, following the basic java coding standard.

mjyan02 commented on src/main/java/parakeet/Ui.java:
```diff
@@ -0,0 +1,51 @@
+        return command;
+    }
+
+    public void colseScanner() {
```
Is this a typo? The method name was intended to be closeScanner?

mjyan02 commented on src/main/java/parakeet/Parser.java:
```diff
@@ -0,0 +1,185 @@
+package parakeet;
+
+import parakeet.command.*;
```
Import classes should always be listed explicitly. There are other files that could use the same edit.

mjyan02 commented on src/main/java/parakeet/command/DeleteCommand.java:
```diff
@@ -0,0 +1,31 @@
+import parakeet.task.Deadline;
+import parakeet.task.Task;
+
+
```
Is there an extra line here? There are other files that seem to have extra lines added.

mjyan02 commented on src/main/java/parakeet/command/DeleteCommand.java:
```diff
@@ -0,0 +1,31 @@
+
+public class DeleteCommand extends Command {
+    private int taskIndex;
+    public DeleteCommand(int taskIndex) {
```
Should an extra line be added here to separate the variable and constructor?

mjyan02 commented on src/main/java/parakeet/command/EventCommand.java:
```diff
@@ -0,0 +1,42 @@
+import parakeet.Ui;
+import parakeet.task.Event;
+import parakeet.task.Task;
+public class EventCommand extends AddCommand {
```
Should an extra line be added here to separate the import statements and the class? There are a few other files that could use the same edits

mjyan02 commented on src/main/java/parakeet/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    /**
+     * Executes the command to add a Deadline task to the task list.
```
I like how you used line breaks to separate the header and ensure readability.

mjyan02 commented on src/main/java/parakeet/command/EventCommand.java:
```diff
@@ -0,0 +1,42 @@
+        Task newEvent = new Event(false, description, startTime, endTime);
+        taskList.add(newEvent);
+
+        String response = "Got it. I've added this task: \n" + newEvent.toString() + "\n"
```
I like how you placed the line breaks to improve readability.

mjyan02 commented on src/test/java/parakeet/StorageTest.java:
```diff
@@ -0,0 +1,63 @@
+        // Arrange: Add tasks
+        Task todo = new Todo(false, "Read book");
+        Task deadline = new Deadline(true, "Submit assignment"
+                , LocalDateTime.of(2024, 1, 30, 23, 59));
```
Line break should be after comma. Have the comma in the previous line?

mjyan02 commented on src/test/java/parakeet/StorageTest.java:
```diff
@@ -0,0 +1,63 @@
+package parakeet;
```
Perhaps it might be better to separate package and the various import statements to include readability

mjyan02 reviewed (commented):
Naming conventions are good. Some missing line breaks or extra line breaks here and there that could be edited. Overall, the code is still very readable. Keep up the good work!

LsmnBmnc commented on src/main/java/parakeet/command/AddCommand.java:
```diff
@@ -0,0 +1,26 @@
+package parakeet.command;
```
Do we need this class?

advaypakhale commented on src/main/java/parakeet/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+        return str;
+    }
+
+    public TaskList find(String keyword) {
```
perhaps could change find to findTasks and some similar other changes for other method names to make clear what the purpose of the method is?

advaypakhale commented on src/main/java/parakeet/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,39 @@
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
+        Task newDeadline = new Deadline(false, description, deadline);
+        taskList.add(newDeadline);
+        String response = "Got it. I've added this task: \n" + newDeadline.toString() + "\n"
```
perhaps you could use String.format method here instead of concatenation to make the code more extensible and look a bit cleaner?

advaypakhale commented on src/main/java/parakeet/command/FindCommand.java:
```diff
@@ -0,0 +1,22 @@
+import parakeet.TaskList;
+import parakeet.Ui;
+
+public class FindCommand extends Command {
+
+    private String keyword;
+
+    public FindCommand(String keyword) {
+        this.keyword = keyword;
+    }
+
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
```
I like your consistent spacing between methods, attributes, imports etc.

advaypakhale commented on src/main/java/parakeet/task/Task.java:
```diff
@@ -0,0 +1,38 @@
+        return this.description.contains(keyword);
+    }
+
+    public void unDone() {
+        this.isDone = false;
+    }
```
perhaps unDone seems a bit clunky in terms of naming, could it be something clearer like markIncomplete?

advaypakhale commented on src/main/java/parakeet/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+     */
+    public String convertToFileFormat() {
+        String str = "";
+        for (int i =0; i < taskList.size(); i++) {
```
some spacing issue here for int i = 0

advaypakhale reviewed (commented):
Overall, awesome job! Just some nits about spelling and naming of methods etc, but easy fixes overall

carolineYQR committed:
Refactor Parser class to improve code quality

The method for parsing commands is too long and contains duplicated
validatoin logic, deep nested conditionals. This makes the code harder
to read.

Let's refactor the method by:
*Extract statements into method
*Using guard clauses to reduce nesting
*Remove unused variable and class

These improvements enhance readability and code quality.

carolineYQR committed:
Add assertions in the methods

Certain assumptions must be hold true on key assumptions.

However, these assumptionsare not explicitly enforced, which could
lead to unexpected errors during development.

Let's add assertions to validate:
* The command string before processing
* The extracted event description
* The correctness of regex matching

These assertions serve as internal sanity checks, improving code
reliability without affecting production behavior.

carolineYQR committed:
Add feature stream to the method

caroline1233456 committed:
Merge pull request #1 from caroline1233456/branch-A-Assertions

Add assertions in the methods

carolineYQR committed:
Resolve conflict between master and this branch

carolineYQR committed:
Resolve conflicts between master and this branch

caroline1233456 committed:
Merge pull request #2 from caroline1233456/branch-A-CodeQuality

Merge branch-A-CodeQuality to master

carolineYQR committed:
Resolve conflicts between master and this branch

caroline1233456 committed:
Merge pull request #3 from caroline1233456/branch-A-Streams

Merge branch-A-Streams to master

carolineYQR committed:
Add Extension C-DetectDuplicates

caroline1233456 committed:
Merge pull request #4 from caroline1233456/branch-BCD-Extension

Merge branch-BCD-Extension to master

carolineYQR committed:
Add a representative screenshot to the docs folder

carolineYQR committed:
Add a brief User Guide

carolineYQR committed:
Finalize ip

carolineYQR committed:
Improve Readme.md

carolineYQR committed:
Add exit command

carolineYQR committed:
Refactor Gui to improve code quality

carolineYQR committed:
Update UserGuide

carolineYQR committed:
Refactor the code

carolineYQR committed:
Debug bye command
